src/commonMain/kotlin/fr/acinq/lightning/payment/OutgoingPaymentPacket.kt

```diff
@@ -81,8 +82,20 @@ object OutgoingPaymentPacket {
             }
             Triple(amount + hop.fee(amount), expiry + hop.cltvExpiryDelta, listOf(payload) + payloads)
         }
+        var payloads = initialPayloads
         val nodes = hops.map { it.nextNodeId }
-        val onion = buildOnion(nodes, payloads, invoice.paymentHash, payloadLength = null)
+        var onion = buildOnion(nodes, payloads, invoice.paymentHash, payloadLength = null)
+        // Ensure that this onion can fit inside the outer 1300 bytes onion.
+        while (onion.packet.payload.size() > 1142) {
```

Comment:
Where does 1142 come from? We probably don't want to explicitly detail each byte (especially since there is a risk that it becomes outdated if something changes in the outer onion), so we should probably:

- take some safety margin to keep some buffer for future additional data and restrict to 1000 bytes
- add a comment explaining that the outer onion fields needs ~150 bytes and we're leaving some buffer space available

Reply:
Done